src/Matcher.js

```diff
@@ -125,10 +125,18 @@ export default class Matcher {
       params[name] = value && decodeURIComponent(value);
     });
 
-    return {
-      params,
-      remaining: pathname.slice(match[0].length),
-    };
+    let remaining = pathname.slice(match[0].length);
+    if (this.endsWithDelimiter(match[0])) {
```

Comment:
This is sort of interesting and comes up for the "named child routes" bit. If you had a route definition like:

```js
<Route path="foo">
  <Route path="*" />
</Route>
```

Would you expect the child route to match on a path of `/foo`? What about on a path of `/foo/`?

For something like:

```js
<Route path="foo">
  <Route path=":param" />
</Route>
```

The answer will be "no" for both.